commit 8ddf7c3d04b8925c6fcd29695e63d4bab3a47878
Author: Andrea Iacovitti <aiacovitti@libero.it>
Date:   Tue Dec 10 18:33:34 2013 +0100

    Fix network status determination: m_networkStatus is a bool,
    and remove it as it is unused.

diff --git a/plasma/generic/dataengines/geolocation/geolocation.cpp b/plasma/generic/dataengines/geolocation/geolocation.cpp
index f5dc4b8..65c376a 100644
--- a/plasma/generic/dataengines/geolocation/geolocation.cpp
+++ b/plasma/generic/dataengines/geolocation/geolocation.cpp
@@ -25,8 +25,7 @@
 static const char SOURCE[] = "location";
 
 Geolocation::Geolocation(QObject* parent, const QVariantList& args)
-    : Plasma::DataEngine(parent, args),
-      m_networkStatus(false)
+    : Plasma::DataEngine(parent, args)
 {
     Q_UNUSED(args)
     setMinimumPollingInterval(500);
@@ -39,8 +38,6 @@ Geolocation::Geolocation(QObject* parent, const QVariantList& args)
 
 void Geolocation::init()
 {
-    m_networkStatus = Solid::Networking::status();
-
     //TODO: should this be delayed even further, e.g. when the source is requested?
     const KService::List offers = KServiceTypeTrader::self()->query("Plasma/GeolocationProvider");
     QVariantList args;
@@ -110,9 +107,8 @@ bool Geolocation::sourceRequestEvent(const QString &name)
 void Geolocation::networkStatusChanged()
 {
     kDebug() << "network status changed";
-    m_networkStatus = Solid::Networking::status();
-    if ((m_networkStatus == Solid::Networking::Connected) ||
-        (m_networkStatus == Solid::Networking::Unknown)) {
+    const Solid::Networking::Status netStatus = Solid::Networking::status();
+    if ((netStatus == Solid::Networking::Connected) || (netStatus == Solid::Networking::Unknown)) {
         updatePlugins(GeolocationProvider::NetworkConnected);
     }
 }
diff --git a/plasma/generic/dataengines/geolocation/geolocation.h b/plasma/generic/dataengines/geolocation/geolocation.h
index 78ec4e3..88acc8b 100644
--- a/plasma/generic/dataengines/geolocation/geolocation.h
+++ b/plasma/generic/dataengines/geolocation/geolocation.h
@@ -53,7 +53,6 @@ class Geolocation : public Plasma::DataEngine
         EntryAccuracy m_accuracy;
         QList<GeolocationProvider *> m_plugins;
         QTimer m_updateTimer;
-        bool m_networkStatus;
 };
 
 K_EXPORT_PLASMA_DATAENGINE(geolocation, Geolocation)
